modules/account-lib/src/coin/dot/transactionBuilder.ts

```diff
@@ -1,57 +1,368 @@
+import { decode, getRegistry } from '@substrate/txwrapper-polkadot';
 import BigNumber from 'bignumber.js';
-import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseTransactionBuilder } from '../baseCoin';
-import { NotImplementedError } from '../baseCoin/errors';
+import { isValidEd25519Seed } from '../../utils/crypto';
+import { BaseTransactionBuilder, TransactionType } from '../baseCoin';
+import { BuildTransactionError, InvalidTransactionError } from '../baseCoin/errors';
 import { BaseAddress, BaseKey } from '../baseCoin/iface';
+import { AddressValidationError } from './errors';
+import { CreateBaseTxInfo, FeeOptions, sequenceId, TxMethod, validityWindow } from './iface';
+import { KeyPair } from './keyPair';
 import { Transaction } from './transaction';
+import { BaseTransactionSchema, SignedTransactionSchema, SigningPayloadTransactionSchema } from './txnSchema';
+import { default as Utils, default as utils } from './utils';
 
 export abstract class TransactionBuilder extends BaseTransactionBuilder {
-  private _transaction: Transaction;
+  protected _transaction: Transaction;
+  protected _keyPair: KeyPair;
+  protected _sender: string;
+
+  protected _blockNumber: number;
+  protected _blockHash: string;
+  protected _genesisHash: string;
+  protected _metadataRpc: string;
+  protected _specVersion: number;
+  protected _transactionVersion: number;
+  protected _specName: PolkadotSpecNameType;
+  protected _chainName: string;
+  protected _nonce: number;
+  protected _tip?: number;
+  protected _eraPeriod?: number;
+  protected _registry: TypeRegistry;
+  protected _method?: TxMethod;
 
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
+    this._transaction = new Transaction(_coinConfig);
+    this.staticsConfig();
+  }
+
+  /**
+   * Sets the address of sending account.
+   *
+   * @param {BaseAddress} address The SS58-encoded address.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  sender({ address }: BaseAddress): this {
+    this.validateAddress({ address });
+    this._sender = address;
+    this._transaction.sender(address);
+    return this;
+  }
+
+  /**
+   * The nonce for this transaction.
+   *
+   * @param {number} nonce
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  sequenceId(nonce: sequenceId): this {
+    const value = new BigNumber(nonce.value);
+    this.validateValue(value);
+    this._nonce = value.toNumber();
+    return this;
+  }
+
+  /**
+   * The tip to increase transaction priority.
+   *
+   * @param {number | undefined} tip optional
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://wiki.polkadot.network/docs/build-transaction-construction
+   */
+  tip(tip: FeeOptions): this {
```

Comment:
Add a follow-up GH issue to rename this to `fee` and make sure it can only take the `FeeOptions` parameters the protocol supports